plutus-core/plutus-core/src/PlutusCore/Default/Builtins.hs

```diff
@@ -1363,6 +1389,93 @@ instance uni ~ DefaultUni => ToBuiltinMeaning uni DefaultFun where
         makeBuiltinMeaning
             (\() -> [] @(Data,Data))
             (runCostingFunOneArgument . paramMkNilPairData)
+    -- BLS12_381.G1
+    toBuiltinMeaning _var Bls12_381_G1_add =
+        makeBuiltinMeaning
+            BLS12_381.G1.add
+            (runCostingFunTwoArguments . paramBls12_381_G1_add)
+    toBuiltinMeaning _var Bls12_381_G1_neg =
+        makeBuiltinMeaning
+            BLS12_381.G1.neg
+            (runCostingFunOneArgument . paramBls12_381_G1_neg)
+    toBuiltinMeaning _var Bls12_381_G1_scalarMul =
+        makeBuiltinMeaning
+            BLS12_381.G1.scalarMul
+            (runCostingFunTwoArguments . paramBls12_381_G1_scalarMul)
+    toBuiltinMeaning _var Bls12_381_G1_compress =
+        makeBuiltinMeaning
+            BLS12_381.G1.compress
+            (runCostingFunOneArgument . paramBls12_381_G1_compress)
+    toBuiltinMeaning _var Bls12_381_G1_uncompress =
+        makeBuiltinMeaning
+            uncompressG1
+            (runCostingFunOneArgument . paramBls12_381_G1_uncompress)
+        where uncompressG1 s =
+                  case BLS12_381.G1.uncompress s of
+                    Left _  -> EvaluationFailure
```

Comment:
OK, I've done that now, and added a special error type instead of a Text error.  I wondered whether to do the emission here or in the definition of the Haskell function as is done [here](https://github.com/input-output-hk/plutus/blob/2c733ab870e709fd09a888754fc02ca048334df3/plutus-core/plutus-core/src/PlutusCore/Crypto/Secp256k1.hs#L41), for example,  but eventually decided to just confine the emitter stuff to the point where we actually need it. I'm now tempted to do the same for the division by zero check [here](https://github.com/input-output-hk/plutus/blob/2c733ab870e709fd09a888754fc02ca048334df3/plutus-core/plutus-core/src/PlutusCore/Default/Builtins.hs#L170).